sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -196,24 +225,19 @@ class FrameLessOffsetWindowFunctionFrame(
     offset: Int,
     ignoreNulls: Boolean = false)
   extends OffsetWindowFunctionFrameBase(
-    target, ordinal, expressions, inputSchema, newMutableProjection, offset) {
+    target, ordinal, expressions, inputSchema, newMutableProjection, offset, ignoreNulls) {
 
-  override def prepare(rows: ExternalAppendOnlyUnsafeRowArray): Unit = {
+  override def prepareForDefaultValue(rows: ExternalAppendOnlyUnsafeRowArray): Unit = {
+    // During the write phase, always access the current window group. So need reset the states.
```

Comment:
`resetStates` is very cheap, why not we always call it at the beginning? `if (absOffset > rows.length)` is not a common case and perf doesn't matter there.

Reply:
SGTM.